src/BuildScriptGenerator/Node/NpmDependencyChecker.cs

```diff
@@ -0,0 +1,59 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Node
+{
+    [Checker(NodeConstants.NpmName)]
+    public class NpmDependencyChecker : IChecker
+    {
+        // Lists packages that should not be used, but were NOT marked as "deprecated" in npm itself.
+        private static readonly IDictionary<string, string> SupersededPackages = new Dictionary<string, string>
```

Comment:
I think  it will be better to move this dictionary as stand alone and independent of language, every time we add a tool/package etc we can just add it there.

Reply:
but it's very dependent of the language - for example, the way we check NPM packages isn't the same as we do for Pip

Reply:
that's true .. but correct me if im  wrong .. what we are doing here is basically scanning the packagejson for each package and checking whether it exists in the dictionary? Will not that be similar for python where we can check the list of packages from reqrmnt.txt and scanning in the dictionary to see if there is any outdated package used by the user?

Also note that it will make our builds longer.

Reply:
question: this checker doesn't care about dev dependencies right?

Reply:
1. Why would it make the builds longer? It's a simple dict search (O(1))
2. Yes, it checks devDependencies as well (line 32)

Reply:
1. yes search will be O(1), but scanning through all the dependencies will be O(n), also list addition adds O(n) once it crosses capacity, so approx. O(n) as a whole. Anyway  my point is to scan the dev dependency only when we need it.

Reply:
> only when we need it
A user might run into issues with these packages even when building locally (without oryx); thus, IMO, we should let them know about these (potential) problems even if they won't directly affect the build.
To me, it's just another way of reducing customer frustration, not necessarily improving the Oryx build success rate

Reply:
well, I saw we do dev dependencies for npm install all the time, in that case this is not needed. so ignore this